x-pack/libbeat/common/aws/credentials.go

```diff
@@ -44,134 +42,172 @@ type ConfigAWS struct {
 	}
 
 	var proxy func(*http.Request) (*url.URL, error)
-	if config.ProxyUrl != "" {
-		proxyUrl, err := httpcommon.NewProxyURIFromString(config.ProxyUrl)
+	if beatsConfig.ProxyUrl != "" {
+		proxyUrl, err := httpcommon.NewProxyURIFromString(beatsConfig.ProxyUrl)
 		if err != nil {
-			return AWSConfig, err
+			return awsConfig, err
 		}
 		proxy = http.ProxyURL(proxyUrl.URI())
 	}
 	var tlsConfig *tls.Config
-	if config.TLS != nil {
-		TLSConfig, _ := tlscommon.LoadTLSConfig(config.TLS)
+	if beatsConfig.TLS != nil {
+		TLSConfig, _ := tlscommon.LoadTLSConfig(beatsConfig.TLS)
 		tlsConfig = TLSConfig.ToConfig()
 	}
-	AWSConfig.HTTPClient = &http.Client{
+	awsConfig.HTTPClient = &http.Client{
 		Transport: &http.Transport{
 			Proxy:           proxy,
 			TLSClientConfig: tlsConfig,
 		},
 	}
-	return AWSConfig, nil
+	return awsConfig, nil
 }
 
 // GetAWSCredentials function gets aws credentials from the config.
 // If access keys given, use them as credentials.
 // If access keys are not given, then load from AWS config file. If credential_profile_name is not
 // given, default profile will be used.
 // If role_arn is given, assume the IAM role either with access keys or default profile.
-func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
+func GetAWSCredentials(beatsConfig ConfigAWS) (awssdk.Config, error) {
 	// Check if accessKeyID or secretAccessKey or sessionToken is given from configuration
-	if config.AccessKeyID != "" || config.SecretAccessKey != "" || config.SessionToken != "" {
-		return getAccessKeys(config), nil
+	if beatsConfig.AccessKeyID != "" || beatsConfig.SecretAccessKey != "" || beatsConfig.SessionToken != "" {
+		return getConfigForKeys(beatsConfig), nil
 	}
 
-	return getSharedCredentialProfile(config)
+	return getConfigSharedCredentialProfile(beatsConfig)
 }
 
-func getAccessKeys(config ConfigAWS) awssdk.Config {
-	awsConfig := defaults.Config()
+// getConfigForKeys creates a default AWS config and adds a CredentialsProvider using the provided Beats config.
+// Provided config must contain an accessKeyID, secretAccessKey and sessionToken to generate a valid CredentialsProfile
+func getConfigForKeys(beatsConfig ConfigAWS) awssdk.Config {
+	logger := logp.NewLogger("getConfigForKeys")
+
+	config := awssdk.NewConfig()
 	awsCredentials := awssdk.Credentials{
-		AccessKeyID:     config.AccessKeyID,
-		SecretAccessKey: config.SecretAccessKey,
+		AccessKeyID:     beatsConfig.AccessKeyID,
+		SecretAccessKey: beatsConfig.SecretAccessKey,
 	}
 
-	if config.SessionToken != "" {
-		awsCredentials.SessionToken = config.SessionToken
+	if beatsConfig.SessionToken != "" {
+		awsCredentials.SessionToken = beatsConfig.SessionToken
 	}
 
-	awsConfig.Credentials = awssdk.StaticCredentialsProvider{
+	config.Credentials = credentials.StaticCredentialsProvider{
 		Value: awsCredentials,
 	}
 
-	return awsConfig
+	// Assume IAM role if iam_role config parameter is given
```

Comment:
isn't this taken care of from https://github.com/elastic/beats/pull/31224/files#diff-87bf43e29ebf053e042e3327ebcf103820d00d47a4265f883f26be23e8c7cbf2R56-R58?